test-apps/crypto-trading/queries/src/test/java/org/elasticsoftware/cryptotrading/CryptoTradingQueryApiTest.java

```diff
@@ -646,4 +646,160 @@ public void initialize(ConfigurableApplicationContext applicationContext) {
+                .expectStatus().is2xxSuccessful()
+                .expectBody(AccountOutput.class)
+                .returnResult()
+                .getResponseBody();
+
+        assertThat(accountOutput).isNotNull();
+        String userId = accountOutput.userId();
+
+        // Add BTC balance
+        webTestClient.post()
+                .uri("/v1/accounts/" + userId + "/wallet/balances")
+                .bodyValue(new CreateBalanceInput("BTC"))
+                .exchange()
+                .expectStatus().is2xxSuccessful();
+
+        // Credit BTC balance
+        webTestClient.post()
+                .uri("/v1/accounts/" + userId + "/wallet/balances/BTC/credit")
+                .bodyValue(new CreditWalletInput(new BigDecimal("2.0")))
+                .exchange()
+                .expectStatus().is2xxSuccessful();
+
+        // Add EUR balance
+        webTestClient.post()
+                .uri("/v1/accounts/" + userId + "/wallet/balances")
+                .bodyValue(new CreateBalanceInput("EUR"))
+                .exchange()
+                .expectStatus().is2xxSuccessful();
+
+        // Place first sell order
+        SellOrderInput sellOrderInput1 = new SellOrderInput("BTC-EUR", new BigDecimal("0.5"), "sell-ref-1");
+        OrderOutput orderOutput1 = webTestClient.post()
+                .uri("/v1/accounts/" + userId + "/orders/sell")
+                .bodyValue(sellOrderInput1)
+                .exchange()
+                .expectStatus().is2xxSuccessful()
+                .expectBody(OrderOutput.class)
+                .returnResult()
+                .getResponseBody();
+
+        assertThat(orderOutput1).isNotNull();
+        assertThat(orderOutput1.orderId()).isNotNull();
+
+        // Place second sell order
+        SellOrderInput sellOrderInput2 = new SellOrderInput("BTC-EUR", new BigDecimal("0.3"), "sell-ref-2");
+        OrderOutput orderOutput2 = webTestClient.post()
+                .uri("/v1/accounts/" + userId + "/orders/sell")
+                .bodyValue(sellOrderInput2)
+                .exchange()
+                .expectStatus().is2xxSuccessful()
+                .expectBody(OrderOutput.class)
+                .returnResult()
+                .getResponseBody();
+
+        assertThat(orderOutput2).isNotNull();
+        assertThat(orderOutput2.orderId()).isNotNull();
+
+        // Get open orders
+        webTestClient.get()
+                .uri("/v1/accounts/" + userId + "/orders")
+                .exchange()
+                .expectStatus().is2xxSuccessful()
+                .expectBody()
+                .jsonPath("$.userId").isEqualTo(userId)
+                .jsonPath("$.openSellOrders").isArray()
+                .jsonPath("$.openSellOrders.length()").isEqualTo(2)
+                .jsonPath("$.openSellOrders[0].orderId").exists()
+                .jsonPath("$.openSellOrders[0].market.id").isEqualTo("BTC-EUR")
+                .jsonPath("$.openSellOrders[0].quantity").isEqualTo(0.5)
```

Comment:
the quantity should be a String ("0.5") and not a double. using doubles is dangerous in financial applications. Change to String and fix any code that fails due to this test